File: projects/Grocery-Order-System-Java/RotisserieChicken.java
/**
 * This class represents a rotisserie chicken.
 * It can be hot or cold.
 * 
 * @author sanskar Luitel
 * @version Project 10
 * Lab Section Tue/Thur 1:05-2:45PM
 */
public class RotisserieChicken implements GroceryItem {

    private boolean hot;

    public RotisserieChicken(boolean h) {
        hot = h;
    }

    public String getName() {
        return hot ? "Hot Rotisserie Chicken" : "Rotisserie Chicken";
    }

    public double getFullPrice() {
        return hot ? 7.99 : 6.99;
    }

    public String toString() {
        return String.format("%s, price: $%.2f",
                getName(), getFullPrice());
    }
}

File: projects/Grocery-Order-System-CSharp/proj11/RotisserieChicken.cs
public class RotisserieChicken : IGroceryItem
{
    private bool _hot;

    public RotisserieChicken(bool hot) => _hot = hot;

    public string Name
    {
    get
    {
        if (_hot)
            return "Hot Rotisserie Chicken";
        else
            return "Rotisserie Chicken";
    }
    }


    public double FullPrice
    {
    get
    {
        if (_hot)
            return 7.99;
        else
            return 6.99;
    }
    }


    public override string ToString() => $"{Name}, price: ${FullPrice:F2}";
}

